src/core/experimental/memutils.d

```diff
@@ -0,0 +1,332 @@
+            store16i_sse(d, xmm0);
+            d += 16 - rem;
+            n -= 16 - rem;
+            // Move in blocks of 32.
+            // TODO(stefanos): Experiment with differnt sizes.
+            if (n >= 32)
+            {
+                // Align to (previous) multiple of 32. That does something invisible to the code,
+                // but a good optimizer will avoid a `cmp` instruction inside the loop. With a
+                // multiple of 32, the end of the loop can be (if we assume that `n` is in RDX):
+                // sub RDX, 32;
+                // jge START_OF_THE_LOOP.
+                // Without that, it has to be:
+                // sub RDX, 32;
+                // cmp RDX, 32;
+                // jge START_OF_THE_LOOP
+                // NOTE, that we align on a _previous_ multiple (for 37, we will go to 32). That means
+                // we have somehow to compensate for that, which is done at the end of this function.
+                n &= -32;
+                do
+                {
+                    store32i_sse(d, xmm0);
+                    // NOTE(stefanos): I tried avoiding this operation on `d` by combining
+                    // `d` and `n` in the above loop and going backwards. It was slower in my benchs.
+                    d += 32;
+                    n -= 32;
+                } while (n >= 32);
+            }
+            // Compensate for the last (at most) 32 bytes.
+            store32i_sse(temp-0x10, xmm0);
+        }
+    }
+}
+
+void Dmemset_naive(void *dst, const ubyte val, size_t n)
+{
+    ubyte *d = cast(ubyte*)dst;
+    for (size_t i = 0; i != n; ++i)
+    {
+        d[i] = val;
+    }
+}
+
+// NOTE(stefanos):
+// Range-checking is not needed since the user never
+// pass an `n` (byte count) directly.
+
+// Copied from std.traits
+import core.internal.traits: Unqual;
+
+package template ModifyTypePreservingTQ(alias Modifier, T)
+{
+    static if (is(T U ==          immutable U)) alias ModifyTypePreservingTQ =          immutable Modifier!U;
+    else static if (is(T U == shared inout const U)) alias ModifyTypePreservingTQ = shared inout const Modifier!U;
+    else static if (is(T U == shared inout       U)) alias ModifyTypePreservingTQ = shared inout       Modifier!U;
+    else static if (is(T U == shared       const U)) alias ModifyTypePreservingTQ = shared       const Modifier!U;
+    else static if (is(T U == shared             U)) alias ModifyTypePreservingTQ = shared             Modifier!U;
+    else static if (is(T U ==        inout const U)) alias ModifyTypePreservingTQ =        inout const Modifier!U;
+    else static if (is(T U ==        inout       U)) alias ModifyTypePreservingTQ =              inout Modifier!U;
+    else static if (is(T U ==              const U)) alias ModifyTypePreservingTQ =              const Modifier!U;
+    else                                             alias ModifyTypePreservingTQ =                    Modifier!T;
+}
```

Comment:
If this is copied from std.traits, please move them to `core.internal.traits`

Reply:
I think that `std.traits` includes `core.internal.traits`. If I move them there, wouldn't that create problem?

Reply:
No. On the contrary. You can remove the implementation from `std.traits` after moving them there to avoid a duplicate implementation.
Example: https://github.com/dlang/phobos/pull/6847

Reply:
Yes, but to do that, wouldn't those 2 have to be done at the same time?
Actually, first merge the changes in `core.internal.traits`, then a PR to phobos.
Otherwise, I guess phobos will break.
If that's the case, I guess this means that I have to do a different PR for the move in `core.internal.traits`.

Reply:
> Actually, first merge the changes in core.internal.traits, then a PR to phobos.

Yes, but things won't break if you add them in `core.internal.traits`. 

> If that's the case, I guess this means that I have to do a different PR for the move in core.internal.traits.

It can be done directly here as this way we know that these traits are actually used. Though that's your choice.

Reply:
Ok, good.
> It can be done directly here as this way we know that these traits are actually used. Though that's your choice.
I agree, I proposed a different PR just because this PR might never be accepted, but ok, I'll do it here.